src/sdk/pynni/nni/tuner.py

```diff
@@ -17,31 +17,128 @@
 # DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT
 # OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 # ==================================================================================================
+
+"""
+Tuner is an AutoML algorithm, which generates a new configuration for the next try.
+A new trial will run with this configuration.
+
+See :class:`Tuner`' specification and ``docs/en_US/tuners.rst`` for details.
+"""
+
 import logging
 
 import nni
 
 from .recoverable import Recoverable
 
+__all__ = ['Tuner']
+
 _logger = logging.getLogger(__name__)
 
 
 class Tuner(Recoverable):
+    """
+    Tuner is an AutoML algorithm, which generates a new configuration for the next try.
+    A new trial will run with this configuration.
+
+    This is the abstract base class for all tuners.
+    Tuning algorithms should derive this class and override :meth:`update_search_space`, :meth:`receive_trial_result`,
+    as well as :meth:`generate_parameters` or :meth:`generate_multiple_parameters`.
+
+    After initializing, NNI will first call :meth:`update_search_space` to tell tuner the feasible region,
+    and then call :meth:`generate_parameters` one or more times to request for hyper-parameter configurations.
```

Comment:
Better mention `generate_parameter` here.

Reply:
Do you mean `generate_parameters`? It's already there.

Reply:
I mean `generate_multiple_parameters`. I think users should have a general idea about which method (multiple or not multiple) to override at the very beginning. You could move the explanation in `gen_mul_param` to here. Putting it in tutorials is better, but... never mind.